### PR TITLE
Upgrade TypeScript dependency to 1.8-beta (including baseurl resolution)

### PR DESCRIPTION
**Pull request by mprobst**
This is WIP and doesn't quite compile yet.

### CODE REVIEW

**Comment by mprobst** [comment]
@vikerman PTAL, this should be ready now, thanks to the help of @vladima.

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "minimist": "^1.1.1",
     "source-map": "^0.4.2",
     "source-map-support": "^0.3.1",
-    "typescript": "1.7.3"
+    "typescript": "mprobst/TypeScript#pathMappingModuleResolution"
```
Comment (vikerman) [inline package.json]: Should be 1.8?

Comment (mprobst) [inline package.json]: This matches the version we're using internally at Google. We should upgrade to 1.8 as soon as that is released.

**Comment by vikerman** [comment]
LGTM

**Pull request merged**